File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/mount/PaddockMoveItemRequestMessage.scala





















// Generated on 05/03/2013 10:34:41
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait PaddockMoveItemRequestMessage extends MessageDeserializer {
  val messageId = 6052

  protected trait Message extends BaseMessage with Serializable {
    val messageId = PaddockMoveItemRequestMessage.messageId

    val oldCellId: Short
    val newCellId: Short

    def serialize(buf: Buffer) {
      buf.writeShort(oldCellId)
      buf.writeShort(newCellId)
    }

  }

  protected def create(oldCellId: Short, newCellId: Short): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val oldCellId = buf.readShort()
    val newCellId = buf.readShort()

    create(oldCellId, newCellId)
  }

}

object PaddockMoveItemRequestMessage extends PaddockMoveItemRequestMessage {
  case class MessageImpl(oldCellId: Short, newCellId: Short) extends Message
  type Target = MessageImpl

  override def create(oldCellId: Short, newCellId: Short) = MessageImpl(oldCellId, newCellId)

  def apply(oldCellId: Short, newCellId: Short) = create(oldCellId, newCellId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/PaddockMoveItemRequestMessage.java


// Generated on 05/08/2013 19:37:45
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class PaddockMoveItemRequestMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6052;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public short oldCellId;
    public short newCellId;
    
    public PaddockMoveItemRequestMessage() { }
    
    public PaddockMoveItemRequestMessage(short oldCellId, short newCellId) {
        this.oldCellId = oldCellId;
        this.newCellId = newCellId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(oldCellId);
        buf.writeShort(newCellId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        oldCellId = buf.readShort();
        if (oldCellId < 0 || oldCellId > 559)
            throw new RuntimeException("Forbidden value on oldCellId = " + oldCellId + ", it doesn't respect the following condition : oldCellId < 0 || oldCellId > 559");
        newCellId = buf.readShort();
        if (newCellId < 0 || newCellId > 559)
            throw new RuntimeException("Forbidden value on newCellId = " + newCellId + ", it doesn't respect the following condition : newCellId < 0 || newCellId > 559");
    }
    
}
